src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -8665,7 +8665,7 @@ def where(
                 snowflake_quoted_identifier
             )
             # TODO (SNOW-904421): Other value can fail to cast in snowflake if not compatible type
-            if other_value:
+            if other_value or other_value == 0:
```

Comment:
Could we have same issue if other_value is `False`? Shall we check for `other_value is not None`?

Reply:
Good idea. False worked already because False == 0, but "is not None" is obviously much clearer.